arch/x86/crc32_fold_pclmulqdq.c

```diff
@@ -226,16 +227,16 @@ static void partial_fold(const size_t len, __m128i *xmm_crc0, __m128i *xmm_crc1,
     *xmm_crc3 = _mm_castps_si128(ps_res);
 }
 
-Z_INTERNAL void crc_fold_copy(unsigned int crc0[4 * 5], unsigned char *dst, const unsigned char *src, long len) {
+Z_INTERNAL void crc32_fold_copy_pclmulqdq(crc32_fold *crc, unsigned char *dst, const unsigned char *src, unsigned long len) {
     unsigned long algn_diff;
     __m128i xmm_t0, xmm_t1, xmm_t2, xmm_t3;
     char ALIGNED_(16) partial_buf[16] = { 0 };
 
     /* CRC_LOAD */
-    __m128i xmm_crc0 = _mm_loadu_si128((__m128i *)crc0 + 0);
-    __m128i xmm_crc1 = _mm_loadu_si128((__m128i *)crc0 + 1);
-    __m128i xmm_crc2 = _mm_loadu_si128((__m128i *)crc0 + 2);
-    __m128i xmm_crc3 = _mm_loadu_si128((__m128i *)crc0 + 3);
+    __m128i xmm_crc0 = _mm_loadu_si128((__m128i *)crc->fold + 0);
+    __m128i xmm_crc1 = _mm_loadu_si128((__m128i *)crc->fold + 1);
+    __m128i xmm_crc2 = _mm_loadu_si128((__m128i *)crc->fold + 2);
+    __m128i xmm_crc3 = _mm_loadu_si128((__m128i *)crc->fold + 3);
```

Comment:
Using temporary variable would make this cleaner as it would remove need for repeated casting.
Using unaligned version should also be unnecessary as the structure is guaranteed to be aligned and fold is the first member of the structure.

Reply:
Fixed.